src/cryptography/hazmat/backends/openssl/aead.py

```diff
@@ -84,6 +103,10 @@ def _encrypt(backend, cipher_cls, key, nonce, data, associated_data,
     ctx = _aead_setup(
         backend, cipher_name, key, nonce, None, tag_length, _ENCRYPT
     )
+    # CCM requires us to pass the length of the data before processing anything
+    from cryptography.hazmat.primitives.ciphers.aead import AESCCM
+    if cipher_cls is AESCCM:
```

Comment:
that's obnoxious. can you put in the comment that calling this on other AEAD's is an internal error